content/posts/google_sheets_as_apis.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Accessing Google Sheets Data from Genezio Using a Service Account
+date: 2024-10-15
+tags:
+  - Tutorials
+author: Bogdan Ripa
+linkedIn: https://www.linkedin.com/in/bogdanripa
+thumbnail: /images/full-stack-developer-tools.webp
+preview: "Expose data stored in google sheets as APIs to be used in your Genezio projects."
+description: "Learn how to use Google Sheets APIs to create API endpoints. Follow our detailed guide."
+meta_og_url: "https://genezio.com/blog/google-sheets-as-apis/"
+meta_og_image: "https://genezio.com/images/full-stack-developer-tools.webp"
+customHeader: "White header"
+customFooter: "White footer"
+readTime: 30
+url: /blog/google-sheets-as-apis
+---
+
+Accessing Google Sheets data programmatically can be essential for server-side applications that need to read or manipulate spreadsheet data without human interaction. This tutorial will guide you through accessing a Google Sheet's data from Genezio using a Service Account, which is ideal for server environments.
+
+## Prerequisites
+
+- A **Genezio** account.
+- **Google Account**: Access to the Google Cloud Console.
+- **Existing Google Spreadsheet**: A Google Sheet that you want to access.
+
+## Overview
+
+We'll cover the following steps:
+
+1. **Enable the Google Sheets API** in your Google Cloud project.
+2. **Create a Service Account** and generate a key file.
+3. **Share the Spreadsheet** with the Service Account email.
+4. **Create a Genezio project** starting from an existing template.
+5. **Update the project** with your personal data and credentials.
+6. **Test the project** to verify everything works.
+
+## Step 1: Enable the Google Sheets API
+
+First, enable the Google Sheets API for your project.
+
+1. **Go to the [Google Cloud Console](https://console.cloud.google.com/).**
+
+2. **Create a New Project** (if you don't have one):
+
+   - Click the project dropdown at the top of the page.
+   - Click **New Project**.
+   - Enter a project name (e.g., "Google Sheets Node.js Access").
+   - Click **Create**.
+
+3. **Select Your Project**:
+
+   - Ensure your new project is selected in the project dropdown.
+
+4. **Enable the Google Sheets API**:
+
+   - Navigate to **APIs & Services** > **Dashboard**.
+   - Click **Enable APIs and Services**.
+   - Search for **Google Sheets API**.
+   - Click on **Google Sheets API** and then click **Enable**.
+
```

Comment:
With the API enabled, your project can now interact with Google Sheets data programmatically, preparing for the next steps.